zc_install/sql/updates/mysql_upgrade_zencart_156.sql

```diff
@@ -127,7 +127,16 @@ SELECT products_attributes_id FROM products_attributes WHERE products_id NOT IN
 SELECT products_id
 FROM products));
 
+## alter admin_pages for new product listing pages
+UPDATE admin_pages
+SET language_key = 'BOX_CATALOG_CATEGORY',
+    main_page = 'FILENAME_CATEGORY_PRODUCT_LISTING',
+    display_on_menu = 'N',
+    sort_order = 18
+WHERE page_key = 'categories';
 
+INSERT INTO admin_pages (page_key, language_key, main_page, page_params, menu_key, display_on_menu, sort_order)
+VALUES ('categoriesProductListing', 'BOX_CATALOG_CATEGORIES_PRODUCTS', 'FILENAME_CATEGORY_PRODUCT_LISTING', '', 'catalog', 'Y', 1);
```

Comment:
Why the insertion of a new key, instead of renaming-and-hiding?

Reply:
I always thought the pages had to be in the admin_pages  table, so you can allow or disallow them in the profiles.
That is why I added a new item for the listing, and updated the old item so it represents the category page.